stub-server/src/main/java/org/zanata/rest/service/MockFileResource.java

```diff
@@ -51,14 +62,27 @@
 public class MockFileResource implements FileResource {
     @Override
     public Response acceptedFileTypes() {
+        Set<String> extensions = new HashSet<>();
+        for (DocumentType docType: ProjectType.getSupportedSourceFileTypes(ProjectType.File)) {
+            extensions.addAll(docType.getSourceExtensions());
+        }
+        for (DocumentType docType: ProjectType.getSupportedSourceFileTypes(
+            ProjectType.Gettext)) {
+            extensions.addAll(docType.getSourceExtensions());
+        }
         StringSet entity = new StringSet("");
-        entity.addAll(ProjectType.getSupportedSourceFileTypes(ProjectType.File));
-        entity.addAll(ProjectType
-                .getSupportedSourceFileTypes(ProjectType.Gettext));
+        entity.addAll(extensions);
```

Comment:
What is the reason for the intermediate `Set<String> extensions`, instead of just adding everything directly to `StringSet entity`?